File: ReadBookVip/ReadBookVipJava/backend/ReadBookApi/Models/Chapter.cs
namespace ReadBookApi.Models;

public class Chapter
{
    public long Id { get; set; }
    public long BookId { get; set; }
    public int ChapterNumber { get; set; }
    public string? Title { get; set; }
    public string Content { get; set; } = string.Empty; // HTML content
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    
    // Navigation property
    public Book? Book { get; set; }
}

File: ReadBookVip/ReadBookVipJava/app/src/main/java/com/example/book/model/Chapter.java
package com.example.book.model;

import java.io.Serializable;

public class Chapter implements Serializable {
    private long id;
    private long bookId;
    private int chapterNumber;
    private String title;
    private String content; // HTML content

    public Chapter() {}

    public Chapter(long id, long bookId, int chapterNumber, String title, String content) {
        this.id = id;
        this.bookId = bookId;
        this.chapterNumber = chapterNumber;
        this.title = title;
        this.content = content;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getBookId() {
        return bookId;
    }

    public void setBookId(long bookId) {
        this.bookId = bookId;
    }

    public int getChapterNumber() {
        return chapterNumber;
    }

    public void setChapterNumber(int chapterNumber) {
        this.chapterNumber = chapterNumber;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}

